src/Ledger/Gov.lagda

```diff
@@ -41,6 +49,64 @@ _⊢_⇀⦇_,GOV⦈_     : GovEnv → GovState → List (GovVote ⊎ GovProposal
 \begin{code}[hide]
 open GovActionState
 
+-- convert list of (GovActionID,GovActionState)-pairs to list GovActionID pairs.
+getAidPairsList : GovState → List (GovActionID × GovActionID)
+getAidPairsList aid×states =
+  mapMaybe (λ (aid , aState) → (aid ,_) <$> getHash (prevAction aState)) $ aid×states
+
+-- convert list of (GovActionID,GovActionState)-pairs to SET of GovActionID pairs.
+getAidPairsSet : GovState → ℙ (GovActionID × GovActionID)
+getAidPairsSet aid×states =
+  mapPartial (λ (aid , as) → (aid ,_) <$> getHash (prevAction as)) $ fromList aid×states
+
+-- a list of GovActionID pairs connects the first GovActionID to the second
+_connects_to_ : List (GovActionID × GovActionID) → GovActionID → GovActionID → Set
+[] connects aidNew to aidOld = aidNew ≡ aidOld
+((aid , aidPrev) ∷ s) connects aidNew to aidOld = aid ≡ aidNew × s connects aidPrev to aidOld
+                                                  ⊎ s connects aidNew to aidOld
+--   aid₁ → aid₂ → aid₃ → aid₄ → aid₅
+-- Let     l := (aid₅, aid₄) (aid₄, aid₃) (aid₃, aid₂) (aid₂, aid₁)
+-- Check:  l connects aid₅ to aid₂ :
+--   ((aid₅, aid₄) ∷ s) connects aid₅ to aid₂ = aid₅ ≡ aid₅ × s connects aid₄ to aid₂
+
+[_connects_to_?] : ∀ l aidNew aidOld → Dec (l connects aidNew to aidOld)
+[ [] connects aidNew to aidOld ?] = aidNew ≟ aidOld
+
+[ (aid , aidPrev) ∷ s connects aidNew to aidOld ?] =
+  ((aid ≟ aidNew) ×-dec [ s connects aidPrev to aidOld ?]) ⊎-dec [ s connects aidNew to aidOld ?]
+
+module _ {A : Set}{L : List A}{ℓ : Level}{P : Pred (List A) ℓ} where
+  -- ∃-sublist-⇔ : (l : List A)→ ((fromList l ⊆ fromList L) × P l) ⇔ l ⊆ˡ L × P l
+  -- ∃-sublist-⇔ = {!!}
+
+  ∃?-sublist-⇔ : Dec (∃[ l ](fromList l ⊆ fromList L × P l)) ⇔ Dec (∃[ l ](l ⊆ˡ L × P l))
+  ∃?-sublist-⇔ = {!!}
+
+enactable : EnactState → List (GovActionID × GovActionID) → GovActionID × GovActionState → Set
+enactable e aidPairs = λ (aidNew , as) → case getHashES e (GovActionState.action as) of λ where
+  nothing → ⊤
+  (just aidOld) → ∃[ t ](fromList t ⊆ fromList aidPairs × t connects aidNew to aidOld)
+
+open Equivalence
+
+∃?-connecting-subset : ∀ L {u}{v} → Dec (∃[ l ](l ⊆ˡ L) × (l connects u to v))
+∃?-connecting-subset L {u}{v} with [ L connects u to v ?]
+... | yes p = yes (L , (⊆ˡ-id , p))
+... | no ¬q = {!!}
```

Comment:
This is the main problem that's left to resolve.  I believe a recursive algorithm would work---one that looks at each subpermutation of L of length |L|-1 to find one that connects u to v.